drivers/SmartThings/zigbee-button/src/aqara/init.lua

```diff
@@ -29,74 +29,113 @@ local MFG_CODE = 0x115F
 local MULTISTATE_INPUT_CLUSTER_ID = 0x0012
 local PRESENT_ATTRIBUTE_ID = 0x0055
 
+local COMP_LIST = { "button1", "button2", "all" }
 local FINGERPRINTS = {
-  { mfr = "LUMI", model = "lumi.remote.b1acn02" },
-  { mfr = "LUMI", model = "lumi.remote.acn003" }
+  ["lumi.remote.b1acn02"] = { mfr = "LUMI", btn_cnt = 1 },
+  ["lumi.remote.acn003"] = { mfr = "LUMI", btn_cnt = 1 },
+  ["lumi.remote.b186acn03"] = { mfr = "LUMI", btn_cnt = 1 },
+  ["lumi.remote.b286acn03"] = { mfr = "LUMI", btn_cnt = 3 }
 }
 
 local configuration = {
-    {
-      cluster = MULTISTATE_INPUT_CLUSTER_ID,
-      attribute = PRESENT_ATTRIBUTE_ID,
-      minimum_interval = 3,
-      maximum_interval = 7200,
-      data_type = data_types.Uint16,
-      reportable_change = 1
-    },
-    {
-      cluster = PowerConfiguration.ID,
-      attribute = PowerConfiguration.attributes.BatteryVoltage.ID,
-      minimum_interval = 30,
-      maximum_interval = 3600,
-      data_type = PowerConfiguration.attributes.BatteryVoltage.base_type,
-      reportable_change = 1
-    }
+  {
+    cluster = MULTISTATE_INPUT_CLUSTER_ID,
+    attribute = PRESENT_ATTRIBUTE_ID,
+    minimum_interval = 3,
+    maximum_interval = 7200,
+    data_type = data_types.Uint16,
+    reportable_change = 1
+  },
+  {
+    cluster = PowerConfiguration.ID,
+    attribute = PowerConfiguration.attributes.BatteryVoltage.ID,
+    minimum_interval = 30,
+    maximum_interval = 3600,
+    data_type = PowerConfiguration.attributes.BatteryVoltage.base_type,
+    reportable_change = 1
+  }
 }
 
 local function present_value_attr_handler(driver, device, value, zb_rx)
-    if value.value == 1 then
-        device:emit_event(capabilities.button.button.pushed({state_change = true}))
-    elseif value.value == 2 then
-        device:emit_event(capabilities.button.button.double({state_change = true}))
-    elseif value.value == 0 then
-        device:emit_event(capabilities.button.button.held({state_change = true}))
-    end
+  local end_point = zb_rx.address_header.src_endpoint.value
+  local btn_evt_cnt = FINGERPRINTS[device:get_model()].btn_cnt or 1
+  local evt = capabilities.button.button.held({ state_change = true })
+  if value.value == 1 then
+    evt = capabilities.button.button.pushed({ state_change = true })
+  elseif value.value == 2 then
+    evt = capabilities.button.button.double({ state_change = true })
+  end
+  device:emit_event(evt)
+  if btn_evt_cnt > 1 then
+    device:emit_component_event(device.profile.components[COMP_LIST[end_point]], evt)
```

Comment:
It seems like button events will only be emitted from the `all` component when they come from endpoint 3, but they will always be emitted from the `main` endpoint. That's a bit confusing. 

In the past we've used the `main` component as an any/all component for button presses. It's a pattern with some precedent. Did you avoid that with your profiles because the name is not descriptive?

Reply:
@greens This device appears to have two physical buttons, but it actually functions like a 3-button device. When both buttons are pressed simultaneously, it triggers the button event for the all component, but not for button1 or button2. However, when naming it button3, it would be difficult for users to associate it with pressing simultaneously, so it was suggested to create it as an all component.
The function of the main component is intended to update the main device simultaneously whenever an event occurs at any endpoint, just like other multi-button devices.

If you have better idea about `all` component naming, please let me know